packages/scene-composer/src/store/helpers/serializationHelpers.ts

```diff
@@ -573,6 +585,53 @@ function createDocumentState(
     );
   }
 
+  if (scene.properties) {
+    if (scene.properties[KnownSceneProperty.FogCustomColors]) {
+      scene.properties[KnownSceneProperty.FogCustomColors] = (
+        scene.properties[KnownSceneProperty.FogCustomColors] as string[]
+      ).filter((color) => isValidHexCode(color));
+    }
+    if (scene.properties[KnownSceneProperty.BackgroundCustomColors]) {
+      scene.properties[KnownSceneProperty.BackgroundCustomColors] = (
+        scene.properties[KnownSceneProperty.BackgroundCustomColors] as string[]
+      ).filter((color) => isValidHexCode(color));
+    }
+    if (scene.properties[KnownSceneProperty.GroundCustomColors]) {
+      scene.properties[KnownSceneProperty.GroundCustomColors] = (
+        scene.properties[KnownSceneProperty.GroundCustomColors] as string[]
+      ).filter((color) => isValidHexCode(color));
+    }
+    if (scene.properties[KnownSceneProperty.FogSettings]) {
+      const fogSettings = scene.properties[KnownSceneProperty.FogSettings] as IFogSettings;
+      fogSettings.color = isValidHexCode(fogSettings.color) ? fogSettings.color : DEFAULT_FOG_COLOR;
+      fogSettings.near = fogSettings.near && fogSettings.near > 0 ? fogSettings.near : DEFAULT_FOG_NEAR;
+      fogSettings.far = fogSettings.far && fogSettings.far > 0 ? fogSettings.far : DEFAULT_FOG_FAR;
+      scene.properties[KnownSceneProperty.FogSettings] = fogSettings;
+    }
+    if (scene.properties[KnownSceneProperty.SceneBackgroundSettings]) {
+      const backgroundSettings = scene.properties[
+        KnownSceneProperty.SceneBackgroundSettings
+      ] as ISceneBackgroundSetting;
+      backgroundSettings.color = backgroundSettings.color
+        ? isValidHexCode(backgroundSettings.color)
+          ? backgroundSettings.color
+          : DEFAULT_SCENE_BACKGROUND_COLOR
+        : undefined;
+      scene.properties[KnownSceneProperty.SceneBackgroundSettings] = backgroundSettings;
+    }
+    if (scene.properties[KnownSceneProperty.GroundPlaneSettings]) {
+      const groundPlaneSettings = scene.properties[KnownSceneProperty.GroundPlaneSettings] as IGroundPlaneSettings;
+      groundPlaneSettings.color = groundPlaneSettings.color
```

Comment:
I don't think you need to nest this. If it's `undefined` wouldn't you want to return the default color? 
```
groundPlaneSettings.color = isValidHexCode(groundPlaneSettings.color)
          ? groundPlaneSettings.color
          : DEFAULT_GROUND_PLANE_COLOR

Reply:
No, it's allowed to be undefined because there may be a texture assigned instead of a color and that's okay. So it's only invalid if we try to assign a color that isn't a color.

Reply:
Gracias.